inspirehep/records/api/base.py

```diff
@@ -684,3 +720,173 @@ def _add_file(self, url, original_url=None, filename=None, **kwargs):
 
         metadata["url"] = file_path
         return metadata
+
+    def get_enhanced_data(self, serializer=None):
+        """Prepares serialized record for elasticsearch
+        Args:
+            serializer(Schema): Schema which should be used to serialize/enchance
+        Returns:
+            dict: Data serialized/enhanced by serializer.
+                Do no return data which is not specified in Schema
+
+        """
+        if not self.es_serializer and not serializer:
+            return None
+        if not serializer:
+            serializer_module = importlib.import_module(
+                self.__module__.replace("api", "marshmallow")
+            )
+            serializer = getattr(serializer_module, self.es_serializer)
+        return serializer().dump(self).data
+
+    def get_ui_data(self, serializer=None):
+        """Prepares serialized record for ui
+        Returns:
+            dict: Enhanced record data
+        """
+        if not self.ui_serializer and not serializer:
+            return None
+        if not serializer:
+            serializer_module = importlib.import_module(
+                self.__module__.replace("api", "marshmallow")
+            )
+            serializer = getattr(serializer_module, self.ui_serializer)
+        return serializer().dump(self).data
+
+    def _index(self, delete=None):
+        """Runs index in current process.
+            This is main logic of indexing called inside worker
+            if called from self.index() but when called from self.index()
+            it will run in a worker.
+
+        Args:
+            delete: set to True if record is deleted
+
+        Returns:
+            dict: ES info about indexing
+
+        """
+        if self._schema_type != self.pid_type:
+            logger.warning(
+                f"Record {self.id} is wrapped in {self.__name__} class {self.pid_type} "
+                f"type, but $schema says that this is {self._schema_type} type object!"
+            )
+        if delete:
+            result = InspireRecordIndexer().delete(self)
+        else:
+            result = InspireRecordIndexer().index(self)
+        logger.info(f"Indexing finished: {result}")
+        return result
+
+    def index(self, delete=None):
+        """Reindexing task sent to `index_record`
+        which calls record._index in there.
+
+        Args:
+            delete: set to True if record is deleted,
+                If not set, tries to determine this automagically
+        Returns:
+            celery.result.AsyncResult: Task itself
+        """
+        logger.error(f"Indexing record {self.id}")
+        indexing_args = self._record_index(self, deleted=delete)
+        indexing_args["record_version"] = self.model.version_id
+        task = index_record.delay(**indexing_args)
+        logger.info(f"Record {self.id} send for indexing")
+        return task
+
+    @classmethod
+    def _record_index(cls, record, _id=None, deleted=None):
```

Comment:
do you ever need to call `_record_index` on something that's not an `InspireRecord`? in any case it should be a `staticmethod` at most, as `cls` is not used.